src/main/java/com/example/todolist/API/Restful/RestController/TaskController.java

```
@@ -81,9 +81,31 @@ public ResponseEntity<?> getTaskByTitle(@RequestParam(defaultValue = "", name="t
         var paging = PageRequest.of(pageNo, pageSize);
         var tasks = taskRepository.findByTitleContaining(title, paging);
 
-        var taskResponses = TaskResponse.FromTasksToDtoResponses(tasks.getContent());
+        var taskResponses = TaskResponse.FromTasksToTaskResponses(tasks.getContent());
 
         Page<TaskResponse> pageTaskResponses = new PageImpl<>(taskResponses, paging, tasks.getTotalElements());
         return ResponseEntity.status(200).body(new ResponseWithData<>("",pageTaskResponses));
     }
+
+    @PostMapping("/{id}/complete")
+    public ResponseEntity<?> completeTask(@PathVariable("id") String id, HttpServletRequest request) {
+        var session = authService.getSession(request);
+
+        try {
+            var task = taskService.completeTask(id,session.userId());
+
+            emailSenderService.sendEmailForTaskCompleting(task.user.getEmail(), task.getTitle());
+
+            task.setStatus(TaskStatus.DONE);
+            taskRepository.save(task);
+            TaskResponse taskResponse = TaskResponse.FromTaskToTaskResponse(task);
+
+            return ResponseEntity.status(200).body(new ResponseWithData<>("Task is done", taskResponse));
+
+        } catch (MailException e) {
+            return ResponseEntity.status(500).body(new Response(e.getMessage()));
```

don't handle it like this, if email fails, we can think about retry, loggin, but this time just let it fail